### ISSUE TITLE
Potential problem reported by cppcheck

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/swarm#63]
[cppcheck](http://cppcheck.sourceforge.net/) reports the following error in `algod1.cc` when analyzing swarm's source code:
```sh
cppcheck src/
```

```
[src/algod1.cc:511]: (error) Common realloc mistake: 'global_hits_data' nulled but not freed upon failure
```
cppcheck is an interesting tool to add to our development pipeline. I can also report warnings or portability problems:
```sh
cppcheck --enable=all src/
```

**Comment by torognes** [comment]
Thanks, this is an error. Will fix.

**Comment by torognes** [comment]
That error would cause swarm to ignore out of memory problems in some rare cases. Should not be a serious problem. However, when run with the "enable=all" option I found another more serious problem on line 618 of algod1.cc which could cause sorting of grafted small clusters (when using fastidious) to be wrong. Might lead to some instability.

**Comment by torognes** [comment]
Fixed in 1.2.4.
